### PR TITLE
Deprecate using Buzz

### PR DESCRIPTION
**Pull request by azjezz**

### CODE REVIEW

File: Collector/ApiCollector.php [hunk 1]
```diff
@@ -9,49 +9,59 @@
...
+        {
+            return $this->data;
+        }
+
+        public function getName()
+        {
+            return 'symfony_connect';
+        }
     }
+
```
Comment (nicolas-grekas) [inline Collector/ApiCollector.php]: extra line :)

File: DependencyInjection/CompilerPass/ApiPass.php [hunk 2]
```diff
@@ -5,13 +5,32 @@
...
+use SymfonyCorp\Connect\Api\Api;
 
 class ApiPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if ($container->getParameter('kernel.debug')) {
-            $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
+        if ((new \ReflectionClass(Api::class))->hasMethod('getStringResponse')) {
+            // connect ~5.0
```
Comment (nicolas-grekas) [inline DependencyInjection/CompilerPass/ApiPass.php]: ~5.0.0

File: DependencyInjection/CompilerPass/ApiPass.php [hunk 3]
```diff
@@ -5,13 +5,32 @@
...
+            }
+
+            $container->getDefinition('symfony_connect.collector.api')->clearTags();
+
+            $container->getDefinition('symfony_connect.buzz.message_factory')
+                ->setDeprecated(true, 'Buzz is deprecated since symfonycorp/connect-bundle 5.1 and will be removed in future versions.');
+            $container->getDefinition('symfony_connect.buzz.client')
+                ->setDeprecated(true, 'Buzz is deprecated since symfonycorp/connect-bundle 5.1 and will be removed in future versions.');
+            $container->getDefinition('symfony_connect.buzz')
+                ->setDeprecated(true, 'Buzz is deprecated since symfonycorp/connect-bundle 5.1 and will be removed in future versions.');
```
Comment (nicolas-grekas) [inline DependencyInjection/CompilerPass/ApiPass.php]: these should be defined in the xml file

File: DependencyInjection/CompilerPass/ApiPass.php [hunk 4]
```diff
@@ -5,13 +5,32 @@
...
+                $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
+            }
+        } else {
+            // connect ^5.1
+            if ($container->hasDefinition('http_client')) {
+                $container->getDefinition('symfony_connect.api')->replaceArgument(1, new Reference('http_client'));
+                $container->getDefinition('symfony_connect.oauth_consumer')->replaceArgument(4, new Reference('http_client'));
+            }
+
+            $container->getDefinition('symfony_connect.collector.api')->clearTags();
```
Comment (nicolas-grekas) [inline DependencyInjection/CompilerPass/ApiPass.php]: should be inside the above "if", right?

File: DependencyInjection/CompilerPass/ApiPass.php [hunk 5]
```diff
@@ -5,13 +5,32 @@
...
 class ApiPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if ($container->getParameter('kernel.debug')) {
-            $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
+        if ((new \ReflectionClass(Api::class))->hasMethod('getStringResponse')) {
+            // connect ~5.0
+            if ($container->getParameter('kernel.debug')) {
+                $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
```
Comment (nicolas-grekas) [inline DependencyInjection/CompilerPass/ApiPass.php]: should be called also when 5.1 is used with buzz, right?

File: DependencyInjection/CompilerPass/ApiPass.php (hunk 4 above)
Comment (azjezz) [reply DependencyInjection/CompilerPass/ApiPass.php]: no? in this case `connect` 5.1 is being used, so i don't think we should collect anything

**Comment by nicolas-grekas** [comment]
Thank you @azjezz.

**Pull request merged**